### PR TITLE
Develop: Milestone 5 and 6

### PR DESCRIPTION
**Pull request by oxenprogrammer**
# Project 3: Using Bootstrap Week 3 Project

## Technology Used
- HTML5
- CSS3 (Flexbox, CSS3 Grid, and Floats, Media Queries)
- Twitter Bootstrap 5 Framework

### Description
- This is the clone for the site www.newsweek.com
- Designed with responsiveness for desktop, tablet and mobile

### Project link 
   https://oxenprogrammer.github.io/newsweek-clone/

### CODE REVIEW

**Review by ioilmio — CHANGES_REQUESTED** [review]
## Hello Team @oxenprogrammer @Gracetemitope   👋🏼 

#### The goal of this project is to make a replica of the Newsweek clone, more close to the original = better, and use bootstrap in an effective and efficient way.

### Good job but you need to implement or fix something, let me list what:
### Required mandatory changes 🔴 

- #### In your navigation bar should be present a Newsweek logo image;
- #### Navigation links are not sized correctly;

#### Refer to these screenshots about nav links(the white one) the logo(is centered and use an image), the sign in and subscribe buttons are not in place, you are missing the search link, the weather icon, and the font size is not correct, and responsiveness can be improved;
#### Please fix all of the above;
- #### Original
image

- #### Yours 
image

### Main content section
- It contains all the section you need, but they are poorly rendered, the top section is not scrolling as expected, spaces are not respected typography can be improved, respect margins at the edges of the page;

- #### Yours
image

- #### Original
image

### Look how they are spaced, and not overlapping

- #### In the magazine
- #### Original
image

- #### Yours
image

### "in the magazine" is centered and close to the top section;

### Editor's pick is centered 
- #### Yours 
image

- #### Original
image

- #### Yours 
image

- #### Original
image

Change font and size for this section

### Footer
- #### Yours 
image
- #### Original
image

- ### Icons are also links
- ### Eliminate all the white space after your footer;

# General 

## Always compare your clone to the original to get the more possible close copy for different viewport;
# Fix your linters
## Add to .gitignore files that are not strictly necessary to the project;

## Quick Advice

#### I noticed you are using gulp for your scss compilation into css;

#### If you are using VsCode(highly recommended) I can suggest you an extension to use instead, is called Live sass compiler from the same author of Live server (Ritwick Dey);
#### With this and lil bit of configuration you can get rid of gulp and watch your scss files to be compiled in css in a folder, and also minify it;

#### You are using bootstrap and you should leverage its power, investigate how you can use it to scroll different section in different way as in the original page, and to style your markup directly using bootstrap classes, that provide a full range of options for margin, padding flex, and so on;
#### Use this installation guide to use it in your project [click here](https://getbootstrap.com/) CDN method is recommended;
### Skim trough Bootstrap documentation get an idea of what you can

# Please always refer to the mandatory project's requirements and to the original page in doubts; Fix all issues listed and ask for another review.

#### Thank you.
*For any further information slack me @illuminatosalvatore*

**Review by layzrd — CHANGES_REQUESTED** [review]
### Status: Require changes 🌳 ♲
Hi @, I am  [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR.  👏👏   I appreciate your hard work in replicating newsweek.com. Kindly fix the issues stated in the previous review. 

- [x] Reorder the section as the source page on small devices 
- [x] Fix the "In The Magazine" and "Editors Pick" section header and add the icon.
- [x]  Fix the subscribe and newsletter section alignment and positioning. It's distorted on small and medium screens.
- [x] Fix the social icons in the footer. The icons are misplaced on a small screen.
- [x] Keep the color and text style consistency for the Tech and Science, Culture, Sports titles.
- [x]  Fix the Picks section image size on medium screens
- [x]  Fix the Opinion section text alignments on medium screens.
- [x] Fix Style lint and web hint errors
- [x] Avoid Usage of !important rule

| Your Copy | Source | 
| :---:         |     :---:      |
|  Screen Shot 2020-12-11 at 10 53 09 AM|Screen Shot 2020-12-11 at 12 31 31 PM|
|Screen Shot 2020-12-11 at 10 56 07 AM|Screen Shot 2020-12-11 at 10 57 02 AM|
|Screen Shot 2020-12-11 at 11 31 06 AM|Screen Shot 2020-12-11 at 11 31 32 AM|


 
Once you fix the issues, kindly submit another code review. ♲

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
+
+👤 **Emmanuel Okello**
+
+- GitHub: [@githubhandle](https://github.com/oxenprogrammer)
+- Twitter: [@twitterhandle](https://twitter.com/ox_emmy)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/emanuel-okello-1217b4b3/)
+
+👤 **Ajanaku Temitope Grace**
+
+- GitHub: [@githubhandle](https://github.com/Gracetemitope)
```
Comment (arslanbisharat) [inline README.md]: - [ ] [Optional] You can add more social links here

File: scss/styles.scss [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+	// overflow-y: scroll;
+	// -ms-overflow-style: none;
+	// scrollbar-width: none;
+	
+}
+
+// .main::-webkit-scrollbar {
+// 	display: none;
+// }
+
```
Comment (arslanbisharat) [inline scss/styles.scss]: - [ ] Please remove commented code

File: scss/styles.scss [hunk 3]
```diff
@@ -0,0 +1,772 @@
...
+  }
+  
+  .paragraph {
+    @include paragraph();
+  }
+}
+
+.my-turn,
+.culture {
+  // @include section-spacing();
```
Comment (arslanbisharat) [inline scss/styles.scss]: Please remove commented code

File: scss/styles.scss [hunk 4]
```diff
@@ -0,0 +1,772 @@
...
+      flex: none;
+      flex-grow: 0;
+      flex-shrink: 0;
+      flex-basis: auto;
+      display: block;
+      width: 29px;
+      height: 28px;
+      background: url("https://g.newsweek.com/www/images/logo-n1.svg")
+        center/contain no-repeat #f72210;
+      // background-image: url("https://g.newsweek.com/www/images/logo-n1.svg");
```
Comment (arslanbisharat) [inline scss/styles.scss]: Please remove commented code

File: scss/styles.scss [hunk 5]
```diff
@@ -0,0 +1,772 @@
...
+      font-size: 1.5rem;
+
+      &:hover {
+        color: red;
+        transition: 0.4s all ease;
+      }
+    }
+
+    #black-link {
+      color: black !important;
```
Comment (arslanbisharat) [inline scss/styles.scss]: Please reomve important

File: scss/styles.scss [hunk 6]
```diff
@@ -0,0 +1,772 @@
...
+
+    .sub {
+      background: yellow;
+      border-radius: 15px;
+      font-weight: bolder;
+      border: none;
+    }
+
+    .navbar-toggler {
+      color: rgb(235, 228, 228) !important;
```
Comment (arslanbisharat) [inline scss/styles.scss]: Please remove important

File: scss/styles.scss [hunk 7]
```diff
@@ -0,0 +1,772 @@
...
+          overflow-wrap: break-word;
+        }
+
+        form input,
+        form button {
+          width: 16rem;
+        }
+
+        form button {
+          background-color: $bg-primary !important;
```
Comment (arslanbisharat) [inline scss/styles.scss:656]: Remove important

File: scss/styles.scss [hunk 8]
```diff
@@ -0,0 +1,772 @@
...
+
+      .subscribe-text {
+        h4 {
+          font-weight: bolder;
+          color: $body-color;
+          font-size: 3rem;
+        }
+
+        .btn-subscribe {
+          background-color: $bg-primary !important;
```
Comment (arslanbisharat) [inline scss/styles.scss:605]: Remove important

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**

- [x] Reorder the section as the source page on small devices

 

- [x] Fix the "In The Magazine" and "Editors Pick" section header and add the icon.


- [x] Fix the subscribe and newsletter section alignment and positioning. It's distorted on small and medium screens.


- [x] Fix the social icons in the footer. The icons are misplaced on a small screen.

 


- [x] Fix the Opinion section text alignments on medium screens.

- [x]  Fix Style lint and web hint errors

- [x] Please don't use !important in your code. 

- [x] **VS** should be in the center as present on the original page. 


- [ ] Remove extra white space as present on the original page. 


- [ ] Text IN THE MAGAZINE should be in the center and all the images should have the same height. 


- [ ] Text Editor's Pick should be in the center. 


- [ ] Categories (U.S, World Bussiness, etc) should be bold. 

- [ ] On medium and small screens, the navbar item has more space. Please remove that space. 

- [ ] Footer on small screen should have proper styling as present on the original page. 




**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach me out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

File: scss/styles.scss (hunk 8 above)
Comment (Gracetemitope) [reply scss/styles.scss:605]: Thanks for your reviews. They were really helpful.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested

Hello Developer,

Thank you for making changes.  But you have a bug in your code so I cannot see your project properly. Please remove that bug and submit a new code review. 
Please remove the image from the PR message. You don't need it here. 
VS in the Debate section should have a smaller size. 

When you are done with these changes request another code review. 

Best,
Arslan Bisharat
slack: @arslanbisharat

File: README.md [hunk 9]
```diff
@@ -1,2 +1,69 @@
...
+To get a local copy up and running follow these simple example steps.
+
+
+### Setup
+Checkout the project on your web browser with the provided link.
+
+### Install
+- Clone it
+- run `npm install`
+- run `gulp start`
```
Comment (arslanbisharat) [inline README.md]: We don't need it

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Request changes 😳

I am very happy to see this project and I can see you worked hard on this project. There are only a few issues that we need to address and I am sure that you will end up with a perfect project

_It's okay. We are here to learn. Don't get discouraged. Work even harder. You are almost there.😄_

This is a great opportunity to learn. Microverse is here to help you. 

**Please have a look at the suggestions to get more understanding.**

- [x] VS should be aligned properly as present on the original page. 
chrome-capture (23)

- [ ] Images and text should have the same vertical width. 
chrome-capture (24)

- [ ] 
chrome-capture (25)

- [ ] Remove white space
chrome-capture (26)




**Please make the changes and request a re-review for the given project**

If you get stuck somewhere, don't hesitate to reach me out or schedule s 1:1 meeting with a Full-time TSE. 

Stay safe and give me a smile 😇

### Happy Coding
---------------------------------------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

File: README.md (hunk 9 above)
Comment (Gracetemitope) [reply README.md]: Hiii!! Thanks for the review. In the original, some of these texts overflow...it all depends on the text at the time, it is a dynamic website. We've really done our best please. Thanks alot

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @oxenprogrammer 

- [x] On the main tag, remove the max-width of 85%, only the padding is okay at media screen of 992px on my screen. Or increase the percentage to at least 95%. 

- [x] Also, reduce the font size of your **h4** tag so each card can accommodate more text in the paragraph.

That's it, make these changes then submit another form for a code review.

### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama```

**Comment by oxenprogrammer** [comment]
Thank you for you time.

**Review by layzrd — CHANGES_REQUESTED** [review]
### Status: Require changes 🌳 ♲
Hi @oxenprogrammer and @Gracetemitope, I am  [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR.  👏👏   I appreciate your hard work in replicating newsweek.com. Kindly fix the issues stated in the previous review.

- [ ] Fix the header section navigation is missing the search icon.
- [ ] Align the subscribe and the newsletter section contents to the center on small devices
- [ ] Align the newsletter section contents to the center on medium devices
- [ ] Fix the Newsweek logo size in the footer
- [ ] Fix font styles on font size to make the font as similar as the source


| Your Copy | Source | 
| :---:         |     :---:      |
| Screen Shot 2020-12-12 at 8 11 12 AM | Screen Shot 2020-12-12 at 8 53 42 AM |
|Screen Shot 2020-12-12 at 8 16 53 AM| Screen Shot 2020-12-12 at 8 56 47 AM|
|Screen Shot 2020-12-12 at 8 21 54 AM| Screen Shot 2020-12-12 at 8 57 47 AM|
|Screen Shot 2020-12-12 at 8 28 22 AM| Screen Shot 2020-12-12 at 9 02 05 AM  |
|Screen Shot 2020-12-12 at 8 31 30 AM|  Screen Shot 2020-12-12 at 9 01 22 AM |


 
Once you fix the issues, kindly submit another code review. ♲

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Review by layzrd — APPROVED** [review]
### Status: Approved 🙌 🎇
Hi @Gracetemitope and @oxenprogrammer, I am [Abdulaziz Redi Ali](https://azizali.ml/), your TSE for this PR. Congratulations 🎊 👏, you did very well on this project. 👏👏👏  I appreciate your hard work on replicating newsweek.com.

Now, you can move ahead.

Please don't forget to merge to master. Good luck with the next projects!

Happy coding 👨‍💻
[Abdulaziz Redi Ali](https://azizali.ml/) | [Contact me on Slack](https://microverse-students.slack.com/team/U011SD6JUH2)

**Pull request merged**